### PR TITLE
Add NixOS module

### PR DESCRIPTION
**Pull request by babbaj**

### CODE REVIEW

File: flake.nix
Comment (weewoo22) [inline flake.nix]: I don't think `boot.kernelPatches` is the correct way to provide kernel configuration from a module. None of the modules in Nixpkgs that provide kernel modules use it to provide needed config options. Should use something like https://github.com/NixOS/nixpkgs/blob/a3543230e420f027bb148ad8fe82af2df68f52c6/nixos/modules/hardware/video/amdgpu-pro.nix#L35-L37 & https://github.com/NixOS/nixpkgs/blob/a3543230e420f027bb148ad8fe82af2df68f52c6/nixos/modules/hardware/video/amdgpu-pro.nix#L49-L52

File: flake.nix
Comment (weewoo22) [inline flake.nix]: `KALLSYMS` is already enabled in the default kernel configuration. Nixpkg modules that require `KALLSYMS_ALL` have it enabled alone (see: https://github.com/NixOS/nixpkgs/blob/a3543230e420f027bb148ad8fe82af2df68f52c6/nixos/modules/hardware/video/amdgpu-pro.nix#L39)
```suggestion
```

File: flake.nix
Comment (weewoo22) [inline flake.nix]: More specifically
```suggestion
            name = "memflow-enable-kallsyms-all";
```

File: flake.nix
Comment (weewoo22) [inline flake.nix]: We should leave it up to the user if they want to have the module load on boot or not.
```suggestion
```

File: flake.nix
Comment (weewoo22) [inline flake.nix]: `builtins.currentSystem` is impure
```suggestion
            boot.extraModulePackages = [ (self.memflow-kmod.${pkgs.system} config.boot.kernelPackages.kernel) ];
```

**Review by weewoo22 — CHANGES_REQUESTED** [review]

File: flake.nix
Comment (babbaj) [reply flake.nix]: nixos wiki disagrees 
image

**Comment by weewoo22** [comment]
Okay, I changed my mind and changed it back to a one-size-fits-all config option that both patches the kernel config and auto loads the kmod: `memflow.kvm.enable`. Still not too sure about the kernel config patching since it's really meant for unconditionally applying extra kernel config options not ensuring config options, i.e. it will trigger a kernel build even if you already had the option set from something else. LGTM though. Squash it and I'll merge it once my kernel build finishes and I try it to make sure all is well.

**Comment by babbaj** [comment]
There's no point in removing the options for enabling the kernel patch and auto loading the module. They were defaulted to true so just having enable = true was enough for it to all work

**Comment by weewoo22** [comment]
Okay, we'll keep the `loadModule` & `kernelPatch` which are defaulted to `true`. I just removed them because I thought we don't need that much config granularity really.

**Pull request merged**
